src/buildApplication/buildGlobalContext.ts

```diff
@@ -0,0 +1,38 @@
+import { cliWarn } from '../cli';
+
+/**
+ * Generates the javascript content (as a plain string) that deals with the global scope that needs to be
+ * added to the built worker.
+ *
+ * @returns the plain javascript string that should be added at the top of the the _worker.js file
+ */
+export function generateGlobalJs(): string {
+	const nodeEnv = getNodeEnv();
+	return `globalThis.process = { env: { NODE_ENV: '${nodeEnv}' } };`;
+}
+
+function getNodeEnv(): string {
+	const nextJsNodeEnvs = ['production', 'development', 'test'];
+	const defaultNodeEnv = nextJsNodeEnvs[0];
```

Comment:
Nit: to rely on the ordering of this array is pretty brittle, imo. I'd personally be explicit, or better yet, use an enum.

```suggestion
	const defaultNodeEnv = 'production';
```

Reply:
I've added a unit test which makes sure that the default value is `production` does that help? 🙂

Reply:
I've added a string enum anyways 🙂👍